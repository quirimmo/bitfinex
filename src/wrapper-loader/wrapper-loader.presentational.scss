.generic-abstract-loader-wrapper {
	position: relative;
}

.generic-abstract-loader {
	width: 100%;
	height: 100%;
	background-color: white;
	text-align: center;
	opacity: 0.95;
	z-index: 999999;
}

.generic-abstract-loader img {
	height: 100%;
	max-width: 100%;
}
