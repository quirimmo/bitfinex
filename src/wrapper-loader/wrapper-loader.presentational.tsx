import * as React from 'react';
import './wrapper-loader.presentational.scss';
// @ts-ignore
import * as loader from './dollar-loading.gif';

export interface IWrapperLoaderProps {
	isLoading: boolean;
	imageSrc?: string;
	position: 'fixed' | 'absolute';
}

export class WrapperLoader extends React.Component<IWrapperLoaderProps, any> {
	constructor(props: IWrapperLoaderProps) {
		super(props);
	}

	public render() {
		const imageSrc = this.props.imageSrc || loader;
		return (
			<div className={this.props.isLoading ? 'generic-abstract-loader-wrapper' : ''}>
				{this.props.isLoading && (
					<div style={{ position: this.props.position }} className="generic-abstract-loader">
						<img src={imageSrc} />
					</div>
				)}
				{this.props.children}
			</div>
		);
	}
}
