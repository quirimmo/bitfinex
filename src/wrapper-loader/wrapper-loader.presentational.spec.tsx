import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { WrapperLoader } from './wrapper-loader.presentational';

let wrapperLoader: ShallowWrapper<any, Readonly<{}>, React.Component<any, any, any>>;

describe('Wrapper Loader Presentational', () => {
	beforeEach(() => {
		wrapperLoader = shallow(
			<WrapperLoader position="absolute" isLoading={false}>
				<div>TEST</div>
			</WrapperLoader>
		);
	});

	it('should be defined', () => {
		expect(wrapperLoader).toBeDefined();
	});

	it('should match the snapshot if loading is false', () => {
		expect(wrapperLoader).toMatchSnapshot();
	});

	it('should match the snapshot if loading is true', () => {
		wrapperLoader.setProps({ isLoading: true });
		expect(wrapperLoader).toMatchSnapshot();
	});
});
