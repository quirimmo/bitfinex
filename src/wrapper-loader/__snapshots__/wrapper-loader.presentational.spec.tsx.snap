// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Loader Presentational should match the snapshot if loading is false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WrapperLoader
    isLoading={false}
  >
    <div>
      TEST
    </div>
  </WrapperLoader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div>
          TEST
        </div>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TEST",
        },
        "ref": null,
        "rendered": "TEST",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div>
            TEST
          </div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TEST",
          },
          "ref": null,
          "rendered": "TEST",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Wrapper Loader Presentational should match the snapshot if loading is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WrapperLoader
    isLoading={true}
  >
    <div>
      TEST
    </div>
  </WrapperLoader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="generic-abstract-loader"
        >
          <img
            src={Object {}}
          />
        </div>,
        <div>
          TEST
        </div>,
      ],
      "className": "generic-abstract-loader-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            src={Object {}}
          />,
          "className": "generic-abstract-loader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TEST",
        },
        "ref": null,
        "rendered": "TEST",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="generic-abstract-loader"
          >
            <img
              src={Object {}}
            />
          </div>,
          <div>
            TEST
          </div>,
        ],
        "className": "generic-abstract-loader-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src={Object {}}
            />,
            "className": "generic-abstract-loader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TEST",
          },
          "ref": null,
          "rendered": "TEST",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
